src/main/java/com/iota/iri/network/NeighborRouter.java

```diff
@@ -0,0 +1,840 @@
+                    config.getNeighboringSocketPort());
+            serverSocketChannel.socket().bind(tcpBindAddr);
+            log.info("bound server TCP socket to {}", tcpBindAddr);
+
+            // parse neighbors from configuration
+            initNeighbors();
+
+            // init connections to the wanted neighbors,
+            // this also ensures that the whitelists are updated with the corresponding
+            // IP addresses and domain name mappings.
+            connectToWantedNeighbors();
+
+            long lastReconnectAttempts = System.currentTimeMillis();
+            long reconnectAttemptTimeout = TimeUnit.SECONDS.toMillis(config.getReconnectAttemptIntervalSeconds());
+
+            while (!shutdown.get()) {
+                int selected = selector.select(reconnectAttemptTimeout);
+                if (shutdown.get()) {
+                    break;
+                }
+
+                // reinitialize connections to wanted neighbors
+                long now = System.currentTimeMillis();
+                if (forceReconnectAttempt.get() || now - lastReconnectAttempts > reconnectAttemptTimeout) {
+                    lastReconnectAttempts = now;
+                    forceReconnectAttempt.set(false);
+                    connectToWantedNeighbors();
+                }
+
+                Iterator<SelectionKey> iterator = selector.selectedKeys().iterator();
+                while (iterator.hasNext()) {
+                    try {
+                        SelectionKey key = iterator.next();
+                        if (key.isAcceptable()) {
+                            // new connection from neighbor
+                            ServerSocketChannel srvSocket = (ServerSocketChannel) key.channel();
+                            SocketChannel newConn = srvSocket.accept();
+                            if (newConn == null) {
+                                continue;
+                            }
+                            InetSocketAddress remoteAddr = (InetSocketAddress) newConn.getRemoteAddress();
+                            if (!okToConnect(remoteAddr.getAddress().getHostAddress(), newConn)) {
+                                continue;
+                            }
+                            newConn.socket().setTcpNoDelay(true);
+                            newConn.socket().setSoLinger(true, 0);
+                            newConn.configureBlocking(false);
+                            Neighbor newNeighbor = new NeighborImpl<>(selector, newConn,
+                                    remoteAddr.getAddress().getHostAddress(),
+                                    Neighbor.UNKNOWN_REMOTE_SERVER_SOCKET_PORT, txPipeline);
+                            String domain = ipToDomainMapping.get(remoteAddr.getAddress().getHostAddress());
+                            if (domain != null) {
+                                newNeighbor.setDomain(domain);
```

Comment:
No, because we don't want to trigger a domain lookup from within the selector loop. Additionally, the domain name will be available in the case the neighbor was configured with the node via config or addNeighbor. So if we don't have a domain here, it isn't really necessary to do a domain lookup.